pkg/controller/scyllacluster/status.go

```diff
@@ -1,165 +0,0 @@
-	updateAggregatedStatusFields(&sc.Status)
-
-	klog.V(2).InfoS("Updating status", "ScyllaCluster", klog.KObj(sc))
-
-	_, err := scc.scyllaClient.ScyllaClusters(sc.Namespace).UpdateStatus(ctx, sc, metav1.UpdateOptions{})
-	if err != nil {
-		return err
-	}
-
-	klog.V(2).InfoS("Status updated", "ScyllaCluster", klog.KObj(sc))
-
-	return nil
-}
-
-func (scc *Controller) getScyllaVersion(sts *appsv1.StatefulSet) (string, error) {
-	firstMemberName := fmt.Sprintf("%s-0", sts.Name)
-	firstMember, err := scc.podLister.Pods(sts.Namespace).Get(firstMemberName)
-	if err != nil {
-		return "", err
-	}
-
-	controllerRef := metav1.GetControllerOfNoCopy(firstMember)
-	if controllerRef == nil || controllerRef.UID != sts.UID {
-		return "", fmt.Errorf("foreign pod")
-	}
-
-	version, err := naming.ScyllaVersion(firstMember.Spec.Containers)
-	if err != nil {
-		return "", err
-	}
-
-	return version, nil
-}
-
-// calculateRackStatus calculates a status for the rack.
-// sts and old status may be nil.
-func (scc *Controller) calculateRackStatus(sc *scyllav1.ScyllaCluster, rackName string, sts *appsv1.StatefulSet, oldRackStatus *scyllav1.RackStatus, serviceMap map[string]*corev1.Service) *scyllav1.RackStatus {
-	status := &scyllav1.RackStatus{
-		ReplaceAddressFirstBoot: map[string]string{},
-		Members:                 0,
-		ReadyMembers:            0,
-		AvailableMembers:        pointer.Ptr(int32(0)),
-		UpdatedMembers:          pointer.Ptr(int32(0)),
-		Stale:                   pointer.Ptr(true),
-	}
-
-	// Persist ReplaceAddressFirstBoot.
-	if oldRackStatus != nil && oldRackStatus.ReplaceAddressFirstBoot != nil {
-		status.ReplaceAddressFirstBoot = oldRackStatus.DeepCopy().ReplaceAddressFirstBoot
-	}
-
-	if sts == nil {
-		return status
-	}
-
-	status.Members = *sts.Spec.Replicas
-	status.ReadyMembers = sts.Status.ReadyReplicas
-	status.AvailableMembers = pointer.Ptr(sts.Status.AvailableReplicas)
-	status.UpdatedMembers = pointer.Ptr(sts.Status.UpdatedReplicas)
-	status.Stale = pointer.Ptr(sts.Status.ObservedGeneration < sts.Generation)
-
-	// Update Rack Version
-	if status.Members == 0 {
-		status.Version = sc.Spec.Version
-	} else {
-		version, err := scc.getScyllaVersion(sts)
-		if err != nil {
-			status.Version = ""
-			klog.ErrorS(err, "Can't get scylla version", "ScyllaCluster", klog.KObj(sc))
-		} else {
-			status.Version = version
-		}
-	}
-
-	// Update Upgrading condition
-	desiredRackVersion := sc.Spec.Version
-	actualRackVersion := status.Version
-	if desiredRackVersion != actualRackVersion {
-		controllerhelpers.SetRackCondition(status, scyllav1.RackConditionTypeUpgrading)
-	}
-
-	// Update Scaling Down condition
-	for _, svc := range serviceMap {
-		// Check if there is a decommission in progress
-		if _, ok := svc.Labels[naming.DecommissionedLabel]; ok {
-			// Add MemberLeaving Condition to rack status
-			controllerhelpers.SetRackCondition(status, scyllav1.RackConditionTypeMemberLeaving)
```

Comment:
I don't see a counterpart of this in the updated controller (only in the migration) - was that left out on purpose?